Directory.Build.props

```diff
@@ -1,6 +1,7 @@
 ﻿<Project>
   <PropertyGroup>
-    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <TargetFramework>net5.0</TargetFramework>
```

Comment:
I would propose not always blinding setting this property in the props file. This can cause issues when projects want to multi-target and specify a `<TargetFrameworks>` property. 
Two possible solutions:
1. Setup a condition on this so it is only set when TargetFramework and TargetFrameworks are both unset. 
2. Set the TFM in all csproj directly (personally I like this option because it makes it nice when you want your unit test projects to rev to latest while maintaining back compat for a main lib; but this  is an analyzer project so less picky about that).

Reply:
1. Isn't __1.__ the default? It just works that way? Or do you mean between TargetFramework and TargetFrameworks?
2. This way we can make a default for the entire project.  Which enables upgrades more easily?

Reply:
1. I mean the difference between the singular and plural TFM properties. 
2. In practice I have found very few situation where updating projects like this was desirable. The ease  of upgrade is offset by the headaches caused by TFM conflicts. In many cases it is desirable to have different TFMs. Imagine having a lib where you want compatibility (so netstandard) but to confirm it works on various TFMs you want your unit tests multi-targeting. I find this setup to be more desirable. You can somewhat address it by putting a Directory.Build.props in the sub directories but then you have to handle the recursive loading of the files.

Reply:
I couldn't get <TargetFrameworks> to work with just one target (net5.0). Only with <TargetFramework> so leaving it be, and I think the build works as intended right now, and any project we want to override from the default compiler (5.0) we can include in the csproj.